<launch>
    <!--<node pkg="odas_ros" type="odas_core_node" name="odas_core_node">
        <param name="configuration_path" value="$(find odas_ros)/configuration.cfg"/>
    </node>
    <node pkg="odas_ros" type="odas_server_node.py" name="odas_server_node" output="screen">
        <param name="configuration_path" value="$(find odas_ros)/configuration.cfg"/>
        <param name="sst_frame_id" value="odas"/>

        <remap from="sss" to="audio_signed_16_44100"/>
        <remap from="sst" to="audio_sst"/>
    </node>

    <node pkg="audio_utils" type="raw_file_writer_node.py" name="raw_file_writer_node">
        <param name="output_path" value="/home/marc-antoine/odas_sss_raw"/>

        <remap from="audio_in" to="audio_signed_16_44100"/>
    </node>-->

    <node pkg="audio_utils" type="alsa_capture_node" name="alsa_capture_node">
        <param name="device" value="hw:CARD=1,DEV=0"/>
        <param name="format" value="signed_32"/>
        <param name="channel_count" value="2"/>
        <param name="sampling_frequency" value="44100"/>
        <param name="frame_sample_count" value="256"/>

        <param name="merge" value="true"/>
        <param name="merge_gain" value="5"/>

        <remap from="audio_out" to="audio_signed_32_44100"/>
    </node>

    <!--<node pkg="audio_utils" type="alsa_playback_node" name="alsa_playback_node" output="screen">
        <param name="device" value="hw:CARD=Loopback,DEV=0"/>
        <param name="format" value="signed_32"/>
        <param name="channel_count" value="2"/>
        <param name="sampling_frequency" value="44100"/>
        <param name="frame_sample_count" value="256"/>

        <remap from="audio_in" to="audio_signed_32_44100"/>
    </node>-->

    <!--<node pkg="audio_utils" type="split_channel_node.py" name="split_channel_node">
        <param name="input_format" value="signed_32"/>
        <param name="output_format" value="signed_16"/>
        <param name="channel_count" value="16"/>

        <remap from="audio_in" to="audio_signed_32_44100"/>
        <remap from="audio_out_0" to="audio_signed_16_44100"/>
    </node>-->


    <node pkg="audio_utils" type="resampling_node.py" name="resampling_node">
        <param name="input_format" value="signed_32"/>
        <param name="output_format" value="signed_16"/>
        <param name="channel_count" value="1"/>
        <param name="input_sampling_frequency" value="44100"/>
        <param name="output_sampling_frequency" value="16000"/>
        <param name="input_frame_sample_count" value="256"/>

        <remap from="audio_in" to="audio_signed_32_44100"/>
        <remap from="audio_out" to="audio_signed_16_16000"/>
    </node>

    <node pkg="audio_analysis" type="audio_analysis_node.py" name="audio_analysis_node" output="screen">
        <param name="interval" value="16000"/>
        <param name="voice_probability_threshold" value="0.5"/>

        <remap from="audio_in" to="audio_signed_16_16000"/>
    </node>

    <!--<node pkg="speech_to_text" type="speech_to_text_node.py" name="speech_to_text_node" output="screen">
        <param name="sampling_frequency" value="16000"/>
        <param name="frame_sample_count" value="92"/>
        <param name="request_frame_count" value="20"/>
        <param name="language" value="fr"/>

        <remap from="audio_in" to="audio_signed_16_16000"/>
    </node>-->

    <node pkg="auditory_attention" type="auditory_attention_node.py" name="auditory_attention_node" output="screen">
        <param name="message_rate" value="10"/>
        <param name="sound_rms_attack" value="0.05"/>
        <param name="sound_rms_release" value="0.99"/>
        <param name="sound_rms_presence_threshold" value="0.075"/>

        <param name="robot_name_model_output_index" value="0"/>
        <param name="robot_name_model_probability_threshold" value="0.95"/>        
        <param name="robot_name_model_interval" value="1600"/>
        <param name="robot_name_model_analysis_delay" value="8000"/>
        <param name="robot_name_model_analysis_count" value="5"/>

        <remap from="audio_in" to="audio_signed_16_16000"/>
    </node>

    <!--<node pkg="audio_utils" type="beat_detector_node" name="beat_detector_node">
        <remap from="audio_in" to="audio_signed_16_44100"/>
    </node>-->
    
</launch>
